rclcpp/test/test_node_global_args.cpp

```diff
@@ -33,32 +34,29 @@ class TestNodeWithGlobalArgs : public ::testing::Test
 };
 
 TEST_F(TestNodeWithGlobalArgs, local_arguments_before_global) {
-  auto context = rclcpp::contexts::default_context::get_global_default_context();
-  const std::vector<std::string> arguments = {"__node:=local_arguments_test"};
-  const std::vector<rclcpp::Parameter> initial_values = {};
-  const bool use_global_arguments = true;
-  const bool use_intra_process = false;
-  auto node = rclcpp::Node::make_shared(
-    "orig_name", "", context, arguments, initial_values, use_global_arguments, use_intra_process);
+  auto options = rclcpp::NodeOptions::Builder()
+    .arguments({"__node:=local_arguments_test"})
+    .build();
```

Comment:
I'd like to cast a vote (again) in favor of dropping the builder pattern for the parameter idiom pattern.

I really don't see the benefit of the builder pattern in this case. The only advantage it gives (that I can see) is that the resulting options structure is immutable, but I don't see why that's important or advantages.

On the other hand, the parameter idiom has the advantage of almost always being less code (both on the user's side and on the implementation side), and makes it easier to start with a common options structure, and then customize from there. For example:

```
auto node = Node::make_shared("my_node", get_my_projects_default_node_options().use_intra_process(true));
```

You could enable this by having a version of the constructor for the builder that takes an existing NodeOptions, but it would still require more code.

Maybe you guys have an idea of why an immutable NodeOptions would be nice to have that I can't think of right now, but without that I think the parameter idiom version would be better.

Reply:
I don't think that there is a substantial reason for the immuatable, and I think in retrospect, the parameter idiom looks better.  I'll rework this.